Essentials/src/main/java/com/earth2me/essentials/commands/Commandhat.java

```diff
@@ -33,7 +33,9 @@ protected void run(final Server server, final User user, final String commandLab
                 return;
             }
 
-            if (user.isAuthorized("essentials.hat.prevent-type." + hand.getType().name().toLowerCase())) {
+            final Boolean wildcard = user.isAuthorizedStrict(PERM_PREFIX + "*");
+            final Boolean material = user.isAuthorizedStrict(PERM_PREFIX + hand.getType().name().toLowerCase());
+            if (((wildcard != null && wildcard) && material == null) || ((wildcard != null && wildcard) && material) || ((wildcard == null || !wildcard) && (material != null && material))) {
```

Comment:
```suggestion
            if (((wildcard != null && wildcard) && material == null) || ((wildcard != null && wildcard) && (material != null && material)) || ((wildcard == null || !wildcard) && (material != null && material))) {
```

For future reference:

| Material \ Wildcard | unset | false | true  |
|---------------------|-------|-------|-------|
| unset               | allow | allow | deny  |
| false               | allow | allow | allow |
| true                | deny  | deny  | deny  |